Nakama/Source/NakamaTests/Private/Tests/Realtime/Test_Chat.cpp

```diff
@@ -0,0 +1,248 @@
+
+	// Call the AuthenticateEmail function
+	Client->AuthenticateCustom(FGuid::NewGuid().ToString(), "", true, {}, successCallback, errorCallback);
+
+	// Wait for authentication to complete
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForAsyncQueries(this));
+	//ADD_LATENT_AUTOMATION_COMMAND(FWaitForAsyncQueriesWithTimeout(this, 6.0f));
+
+	// Return true to indicate the test is complete
+	return true;
+}
+
+// Join Chat
+IMPLEMENT_CUSTOM_SIMPLE_AUTOMATION_TEST(JoinGroupChat, FNakamaTestBase, "Nakama.Base.Realtime.Chat.JoinGroupChat", EAutomationTestFlags::ApplicationContextMask | EAutomationTestFlags::ProductFilter)
+inline bool JoinGroupChat::RunTest(const FString& Parameters)
+{
+	// Initiates the test
+	InitiateTest();
+	
+	// Define success callback
+	auto successCallback = [this](UNakamaSession* session)
+	{
+		// Set the session for later use
+		Session = session;
+		
+		// Setup socket:S
+		Socket = Client->SetupRealtimeClient();
+
+		Socket->SetConnectCallback([this]()
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Test Socket connected"));
+
+			auto CreateGroupSuccessCallback = [&](const FNakamaGroup& Group)
+			{
+				UE_LOG(LogTemp, Warning, TEXT("Group created: %s"), *Group.Id);
+
+				// Join the group chat
+
+				auto JoinGroupSuccessCallback = [&](const FNakamaChannel& Channel)
+				{
+					UE_LOG(LogTemp, Warning, TEXT("Joined Group Chat ID: %s"), *Channel.Id);
+					TestTrue("Join Group Chat Test Passed", !Channel.Id.IsEmpty());
+					StopTest();
+				};
+
+				auto JoinGroupErrorCallback = [this](const FNakamaRtError& Error)
+				{
+					UE_LOG(LogTemp, Warning, TEXT("Join Group Chat Error - Join Group. ErrorMessage: %s"), *Error.Message);
+					TestFalse("Join Group Chat Error - Join Group.", true);
+					StopTest();
+				};
+
+				Socket->JoinChat(Group.Id, ENakamaChannelType::GROUP, {}, {}, JoinGroupSuccessCallback, JoinGroupErrorCallback);
+			};
+
+			auto CreateGroupErrorCallback = [this](const FNakamaError& Error)
+			{
+				UE_LOG(LogTemp, Warning, TEXT("Join Group Chat Error - Create Group. ErrorMessage: %s"), *Error.Message);
+				TestFalse("Join Group Chat Error - Create Group.", true);
+				StopTest();
+			};
+
+			FString GroupName = FString::Printf(TEXT("Group chat %s"), *Session->GetAuthToken());
+			Client->CreateGroup(Session, GroupName, TEXT("A group for chatting"), {}, {}, false, {}, CreateGroupSuccessCallback, CreateGroupErrorCallback);
+		});
+		
+		Socket->Connect(Session, true);
+	};
+
+	// Define error callback
+	auto errorCallback = [&](const FNakamaError& Error)
+	{
+		// Test fails if there is an authentication error
+		TestFalse("Authentication Test Failed", true);
+		StopTest();
+	};
+
+	// Call the AuthenticateEmail function
+	Client->AuthenticateCustom(FGuid::NewGuid().ToString(), "", true, {}, successCallback, errorCallback);
+
+	// Wait for authentication to complete
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForAsyncQueries(this));
+	//ADD_LATENT_AUTOMATION_COMMAND(FWaitForAsyncQueriesWithTimeout(this, 6.0f));
```

Comment:
I've noticed these commented out in a few places -- are they necessary

Reply:
No, I was testing with timeouts, all of those can be removed.